test/coverage/datamodel/UuidSerializerTest.cpp

```diff
@@ -23,16 +27,159 @@ class TestFixture : public testing::Test {
+
+	std::string uuid_str =
+	    uprotocol::datamodel::serializer::uuid::AsString::serialize(uuid);
+	EXPECT_EQ(uuid_str, "0000000000000000-0000000000000000");
+
+	uprotocol::v1::UUID deserialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::deserialize(uuid_str);
+	EXPECT_EQ(deserialized_uuid.msb(), 0x0000000000000000);
+	EXPECT_EQ(deserialized_uuid.lsb(), 0x0000000000000000);
+
+	std::vector<uint8_t> uuid_bytes =
+	    uprotocol::datamodel::serializer::uuid::AsBytes::serialize(uuid);
+	std::vector<uint8_t> expected_bytes(16, 0x00);
+	EXPECT_EQ(uuid_bytes, expected_bytes);
+
+	deserialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsBytes::deserialize(
+	        uuid_bytes);
+	EXPECT_EQ(deserialized_uuid.msb(), 0x0000000000000000);
+	EXPECT_EQ(deserialized_uuid.lsb(), 0x0000000000000000);
+}
+
+// Test edge case: maximum values for msb and lsb
+TEST_F(TestUuidSerializer, SerializeDeserializeMaxValues) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0xFFFFFFFFFFFFFFFF);
+	uuid.set_lsb(0xFFFFFFFFFFFFFFFF);
+
+	std::string uuid_str =
+	    uprotocol::datamodel::serializer::uuid::AsString::serialize(uuid);
+	EXPECT_EQ(uuid_str, "ffffffffffffffff-ffffffffffffffff");
+
+	uprotocol::v1::UUID deserialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::deserialize(uuid_str);
+	EXPECT_EQ(deserialized_uuid.msb(), 0xFFFFFFFFFFFFFFFF);
+	EXPECT_EQ(deserialized_uuid.lsb(), 0xFFFFFFFFFFFFFFFF);
+
+	std::vector<uint8_t> uuid_bytes =
+	    uprotocol::datamodel::serializer::uuid::AsBytes::serialize(uuid);
+	std::vector<uint8_t> expected_bytes(16, 0xFF);
+	EXPECT_EQ(uuid_bytes, expected_bytes);
+
+	deserialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsBytes::deserialize(
+	        uuid_bytes);
+	EXPECT_EQ(deserialized_uuid.msb(), 0xFFFFFFFFFFFFFFFF);
+	EXPECT_EQ(deserialized_uuid.lsb(), 0xFFFFFFFFFFFFFFFF);
+}
+
+// Test edge case: zero values for msb and lsb
+TEST_F(TestUuidSerializer, SerializeDeserializeZeroValues) {
+	uprotocol::v1::UUID uuid;
+	uuid.set_msb(0x0000000000000000);
+	uuid.set_lsb(0x0000000000000000);
+
+	std::string uuid_str =
+	    uprotocol::datamodel::serializer::uuid::AsString::serialize(uuid);
+	EXPECT_EQ(uuid_str, "0000000000000000-0000000000000000");
+
+	uprotocol::v1::UUID deserialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsString::deserialize(uuid_str);
+	EXPECT_EQ(deserialized_uuid.msb(), 0x0000000000000000);
+	EXPECT_EQ(deserialized_uuid.lsb(), 0x0000000000000000);
+
+	std::vector<uint8_t> uuid_bytes =
+	    uprotocol::datamodel::serializer::uuid::AsBytes::serialize(uuid);
+	std::vector<uint8_t> expected_bytes(16, 0x00);
+	EXPECT_EQ(uuid_bytes, expected_bytes);
 
-// TODO replace
-TEST_F(TestFixture, SomeTestName) {}
+	deserialized_uuid =
+	    uprotocol::datamodel::serializer::uuid::AsBytes::deserialize(
+	        uuid_bytes);
+	EXPECT_EQ(deserialized_uuid.msb(), 0x0000000000000000);
+	EXPECT_EQ(deserialized_uuid.lsb(), 0x0000000000000000);
+}
```

Comment:
Is this a duplicate of the minimum value test case?

Reply:
yes, thanks for the catch